core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1209,6 +1209,15 @@ public void online(String tableName) throws AccumuloSecurityException, AccumuloE
   public void online(String tableName, boolean wait) throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
     checkArgument(tableName != null, "tableName is null");
     String tableId = Tables.getTableId(context.getInstance(), tableName);
+
+    /**
+     * ACCUMULO-4574 if table is already online return without executing fate operation.
+     */
+    TableState expectedState = Tables.getTableState(context.getInstance(), tableId);
```

Comment:
/me lamenting that we don't have a way to do a non-cached read of ZK and are forced to clear the global cache.

If we push this to the Master, would we have more control over reading from ZK? An unnecessary RPC but avoiding the client-side zk cache clear would be well worth it, IMO.